api/v1/helpers/verbs/doGet.js

```diff
@@ -29,11 +31,41 @@ const httpStatus = require('../../constants').httpStatus;
  *  resource type to retrieve.
  */
 function doGet(req, res, next, props) {
-  u.findByKey(props, req.swagger.params)
-  .then((o) => {
-    res.status(httpStatus.OK).json(u.responsify(o, props, req.method));
-  })
-  .catch((err) => u.handleError(next, err, props.modelName));
+  if (props.cacheEnabled) {
+    const reqParams = req.swagger.params;
+    let cacheKey = reqParams.key.value;
+
+    // cache key is combination of key and fields in specified order.
```

Comment:
Let's say I call `GET /v1/perspectives/ABC?fields=name,rootSubject` and the response is cached. After 45 seconds, another user calls `GET /v1/perspectives/ABC?fields=rootSubject,name`... will they get the cached response or hit the DB?